core/src/main/python/wlsdeploy/util/weblogic_helper.py

```diff
@@ -187,6 +189,14 @@ def get_weblogic_home(self, oracle_home):
             else:
                 wl_home = oracle_home + '/wlserver_10.3'
 
+            # if the path is not a valid directory, try to infer it from the pattern
+            if os.path.isdir(oracle_home) and not os.path.isdir(wl_home):
+                dirs = [f for f in os.listdir(oracle_home) if re.match(r'wlserver.*', f)]
+                if len(dirs) > 0:
+                    wl_home = oracle_home + '/' + dirs[0]
+                else:
+                    raise Exception("Can't find proper folder for WLS")
```

Comment:
throwing an exception here breaks 4 of the WLST unit tests.  We could alter the unit tests to accommodate.  But, is this necessary?  What is the behavior if you simply log an error and let the code continue, versus throwing the exception?  In either case, this string needs to be moved to the i18n properties file.

Reply:
As you noted, a bunch of tests were failing because the ORACLE_HOME is mocked and my code change raised an exception when `wl_home` is not found.
I changed that to return `None` instead and leave it to downstream to check that it is not None instead.